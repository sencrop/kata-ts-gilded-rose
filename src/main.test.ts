import { GildedRose, Item } from "./main";

describe("Gilded Rose", function () {
  const executeUpdateQuality = (
    name: string,
    sellIn: number,
    quality: number
  ): Item => {
    const gildedRose = new GildedRose([new Item(name, sellIn, quality)]);
    return gildedRose.updateQuality()[0];
  };

  it("should update quality", function () {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
    expect(executeUpdateQuality).toVerifyAllCombinations(
      [
        "foo",
        "Aged Brie",
        "Backstage passes to a TAFKAL80ETC concert",
        "Sulfuras, Hand of Ragnaros",
      ],
      [-1, 0, 1, 6, 11],
      [-1, 0, 49, 50, 51]
    );
  });
});
