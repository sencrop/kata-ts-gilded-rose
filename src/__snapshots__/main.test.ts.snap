// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gilded Rose should update quality 1`] = `
Object {
  "Aged Brie,-1,-1": Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": -2,
  },
  "Aged Brie,-1,0": Item {
    "name": "Aged Brie",
    "quality": 2,
    "sellIn": -2,
  },
  "Aged Brie,-1,49": Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": -2,
  },
  "Aged Brie,-1,50": Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": -2,
  },
  "Aged Brie,-1,51": Item {
    "name": "Aged Brie",
    "quality": 51,
    "sellIn": -2,
  },
  "Aged Brie,0,-1": Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": -1,
  },
  "Aged Brie,0,0": Item {
    "name": "Aged Brie",
    "quality": 2,
    "sellIn": -1,
  },
  "Aged Brie,0,49": Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": -1,
  },
  "Aged Brie,0,50": Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": -1,
  },
  "Aged Brie,0,51": Item {
    "name": "Aged Brie",
    "quality": 51,
    "sellIn": -1,
  },
  "Aged Brie,1,-1": Item {
    "name": "Aged Brie",
    "quality": 0,
    "sellIn": 0,
  },
  "Aged Brie,1,0": Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": 0,
  },
  "Aged Brie,1,49": Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": 0,
  },
  "Aged Brie,1,50": Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": 0,
  },
  "Aged Brie,1,51": Item {
    "name": "Aged Brie",
    "quality": 51,
    "sellIn": 0,
  },
  "Aged Brie,11,-1": Item {
    "name": "Aged Brie",
    "quality": 0,
    "sellIn": 10,
  },
  "Aged Brie,11,0": Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": 10,
  },
  "Aged Brie,11,49": Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": 10,
  },
  "Aged Brie,11,50": Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": 10,
  },
  "Aged Brie,11,51": Item {
    "name": "Aged Brie",
    "quality": 51,
    "sellIn": 10,
  },
  "Aged Brie,6,-1": Item {
    "name": "Aged Brie",
    "quality": 0,
    "sellIn": 5,
  },
  "Aged Brie,6,0": Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": 5,
  },
  "Aged Brie,6,49": Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": 5,
  },
  "Aged Brie,6,50": Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": 5,
  },
  "Aged Brie,6,51": Item {
    "name": "Aged Brie",
    "quality": 51,
    "sellIn": 5,
  },
  "Backstage passes to a TAFKAL80ETC concert,-1,-1": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  "Backstage passes to a TAFKAL80ETC concert,-1,0": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  "Backstage passes to a TAFKAL80ETC concert,-1,49": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  "Backstage passes to a TAFKAL80ETC concert,-1,50": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  "Backstage passes to a TAFKAL80ETC concert,-1,51": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  "Backstage passes to a TAFKAL80ETC concert,0,-1": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  "Backstage passes to a TAFKAL80ETC concert,0,0": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  "Backstage passes to a TAFKAL80ETC concert,0,49": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  "Backstage passes to a TAFKAL80ETC concert,0,50": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  "Backstage passes to a TAFKAL80ETC concert,0,51": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  "Backstage passes to a TAFKAL80ETC concert,1,-1": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 2,
    "sellIn": 0,
  },
  "Backstage passes to a TAFKAL80ETC concert,1,0": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 3,
    "sellIn": 0,
  },
  "Backstage passes to a TAFKAL80ETC concert,1,49": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 0,
  },
  "Backstage passes to a TAFKAL80ETC concert,1,50": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 0,
  },
  "Backstage passes to a TAFKAL80ETC concert,1,51": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 51,
    "sellIn": 0,
  },
  "Backstage passes to a TAFKAL80ETC concert,11,-1": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": 10,
  },
  "Backstage passes to a TAFKAL80ETC concert,11,0": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 1,
    "sellIn": 10,
  },
  "Backstage passes to a TAFKAL80ETC concert,11,49": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 10,
  },
  "Backstage passes to a TAFKAL80ETC concert,11,50": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 10,
  },
  "Backstage passes to a TAFKAL80ETC concert,11,51": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 51,
    "sellIn": 10,
  },
  "Backstage passes to a TAFKAL80ETC concert,6,-1": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 1,
    "sellIn": 5,
  },
  "Backstage passes to a TAFKAL80ETC concert,6,0": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 2,
    "sellIn": 5,
  },
  "Backstage passes to a TAFKAL80ETC concert,6,49": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 5,
  },
  "Backstage passes to a TAFKAL80ETC concert,6,50": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 5,
  },
  "Backstage passes to a TAFKAL80ETC concert,6,51": Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 51,
    "sellIn": 5,
  },
  "Sulfuras, Hand of Ragnaros,-1,-1": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": -1,
  },
  "Sulfuras, Hand of Ragnaros,-1,0": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": -1,
  },
  "Sulfuras, Hand of Ragnaros,-1,49": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": -1,
  },
  "Sulfuras, Hand of Ragnaros,-1,50": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 50,
    "sellIn": -1,
  },
  "Sulfuras, Hand of Ragnaros,-1,51": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 51,
    "sellIn": -1,
  },
  "Sulfuras, Hand of Ragnaros,0,-1": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": 0,
  },
  "Sulfuras, Hand of Ragnaros,0,0": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": 0,
  },
  "Sulfuras, Hand of Ragnaros,0,49": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": 0,
  },
  "Sulfuras, Hand of Ragnaros,0,50": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 50,
    "sellIn": 0,
  },
  "Sulfuras, Hand of Ragnaros,0,51": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 51,
    "sellIn": 0,
  },
  "Sulfuras, Hand of Ragnaros,1,-1": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": 1,
  },
  "Sulfuras, Hand of Ragnaros,1,0": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": 1,
  },
  "Sulfuras, Hand of Ragnaros,1,49": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": 1,
  },
  "Sulfuras, Hand of Ragnaros,1,50": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 50,
    "sellIn": 1,
  },
  "Sulfuras, Hand of Ragnaros,1,51": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 51,
    "sellIn": 1,
  },
  "Sulfuras, Hand of Ragnaros,11,-1": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": 11,
  },
  "Sulfuras, Hand of Ragnaros,11,0": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": 11,
  },
  "Sulfuras, Hand of Ragnaros,11,49": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": 11,
  },
  "Sulfuras, Hand of Ragnaros,11,50": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 50,
    "sellIn": 11,
  },
  "Sulfuras, Hand of Ragnaros,11,51": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 51,
    "sellIn": 11,
  },
  "Sulfuras, Hand of Ragnaros,6,-1": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": 6,
  },
  "Sulfuras, Hand of Ragnaros,6,0": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": 6,
  },
  "Sulfuras, Hand of Ragnaros,6,49": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": 6,
  },
  "Sulfuras, Hand of Ragnaros,6,50": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 50,
    "sellIn": 6,
  },
  "Sulfuras, Hand of Ragnaros,6,51": Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 51,
    "sellIn": 6,
  },
  "foo,-1,-1": Item {
    "name": "foo",
    "quality": -1,
    "sellIn": -2,
  },
  "foo,-1,0": Item {
    "name": "foo",
    "quality": 0,
    "sellIn": -2,
  },
  "foo,-1,49": Item {
    "name": "foo",
    "quality": 47,
    "sellIn": -2,
  },
  "foo,-1,50": Item {
    "name": "foo",
    "quality": 48,
    "sellIn": -2,
  },
  "foo,-1,51": Item {
    "name": "foo",
    "quality": 49,
    "sellIn": -2,
  },
  "foo,0,-1": Item {
    "name": "foo",
    "quality": -1,
    "sellIn": -1,
  },
  "foo,0,0": Item {
    "name": "foo",
    "quality": 0,
    "sellIn": -1,
  },
  "foo,0,49": Item {
    "name": "foo",
    "quality": 47,
    "sellIn": -1,
  },
  "foo,0,50": Item {
    "name": "foo",
    "quality": 48,
    "sellIn": -1,
  },
  "foo,0,51": Item {
    "name": "foo",
    "quality": 49,
    "sellIn": -1,
  },
  "foo,1,-1": Item {
    "name": "foo",
    "quality": -1,
    "sellIn": 0,
  },
  "foo,1,0": Item {
    "name": "foo",
    "quality": 0,
    "sellIn": 0,
  },
  "foo,1,49": Item {
    "name": "foo",
    "quality": 48,
    "sellIn": 0,
  },
  "foo,1,50": Item {
    "name": "foo",
    "quality": 49,
    "sellIn": 0,
  },
  "foo,1,51": Item {
    "name": "foo",
    "quality": 50,
    "sellIn": 0,
  },
  "foo,11,-1": Item {
    "name": "foo",
    "quality": -1,
    "sellIn": 10,
  },
  "foo,11,0": Item {
    "name": "foo",
    "quality": 0,
    "sellIn": 10,
  },
  "foo,11,49": Item {
    "name": "foo",
    "quality": 48,
    "sellIn": 10,
  },
  "foo,11,50": Item {
    "name": "foo",
    "quality": 49,
    "sellIn": 10,
  },
  "foo,11,51": Item {
    "name": "foo",
    "quality": 50,
    "sellIn": 10,
  },
  "foo,6,-1": Item {
    "name": "foo",
    "quality": -1,
    "sellIn": 5,
  },
  "foo,6,0": Item {
    "name": "foo",
    "quality": 0,
    "sellIn": 5,
  },
  "foo,6,49": Item {
    "name": "foo",
    "quality": 48,
    "sellIn": 5,
  },
  "foo,6,50": Item {
    "name": "foo",
    "quality": 49,
    "sellIn": 5,
  },
  "foo,6,51": Item {
    "name": "foo",
    "quality": 50,
    "sellIn": 5,
  },
}
`;
